react/week3/todo-list/src/App.js

```diff
@@ -0,0 +1,82 @@
+import { useState, useEffect } from "react";
+import "./App.css";
+import TodoList from "./TodoList";
+import Fetching from "./Fetching";
+import { getNextId } from "./Helper";
+import Timer from "./Timer";
+import PropTypes from "prop-types";
+
+function App() {
+  const [todoText, setTodoText] = useState("");
+  const [deadline, setDeadline] = useState("");
+  const [todos, setTodos] = useState([]);
+
+  useEffect(() => {
+    Fetching().then((data) => {
+      setTodos(data);
+    });
+  }, []);
+
+  const addTodo = () => {
+    if (!todoText) {
+      return alert("Insert new todo");
+    }
+    if (!deadline) {
+      return alert("insert deadline");
+    }
+    //the built-in toISOString method format date to the ISO 8601 format:
+    const currentDay = new Date().toISOString().split("T")[0];
+
+    if (deadline < currentDay) {
+      return alert("deadline is not possible");
+    }
+
+    const nextId = getNextId(todos);
+
+    const newTodo = todos.concat({
+      id: nextId,
+      description: todoText,
+      deadline: deadline,
+    });
+    setTodos(newTodo);
+  };
+  const handleDelete = (id) => {
+    const remainedRows = todos.filter((x) => x.id !== id);
+    setTodos(remainedRows);
+  };
+
+  return (
+    <div className="App">
+      <h2>Todolist</h2>
+      <Timer />
+      <br />
+      <span>Todo description</span>
+      <input
+        type="text"
+        placeholder="Add new todo .."
+        value={todoText}
+        onChange={(e) => setTodoText(e.target.value)}
+      />
+      <br />
+      <span>deadline</span>
+      <input
+        type="date"
+        value={deadline}
+        onChange={(e) => setDeadline(e.target.value)}
+      />
+      <br />
+      <button onClick={addTodo} type="submit">
+        Add todo
+      </button>
+      <TodoList todos={todos} handleDelete={handleDelete} />
+    </div>
+  );
+}
+App.prototype = {
+  deadline: PropTypes.Date,
+  onChange: PropTypes.func,
+  todos: PropTypes.array,
+  id: PropTypes.number,
+  description: PropTypes.string,
+};
```

Comment:
Is this something you were taught in class? If so, just ignore this comment 😄 Normally it is considered quite bad practice to set properties directly on a prototype. If you would like to type your props, you could start to look into TypeScript, although it can be a bit overwhelming in the beginning.